### PR TITLE
fix(calendar): #MC-46 restore right calendar's informations after cancel the modification

### PR DESCRIPTION
**Pull request by adrienFro**
## Describe your changes
cancel is now working for calendar modification
Use updateCalendars to force update of the selected calendar and have informations before modifications and cancel.
## Checklist tests
☑︎ Restore the color of calendar before the start of modification
☑︎ Restore the right information of your calendar in modification view when you go again in it after cancel the modification
## Issue ticket number and link
https://entsupport.gdapublic.fr/browse/MC-46
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/controller.ts [hunk 1]
```diff
@@ -1003,14 +1004,16 @@ export const calendarController = ng.controller('CalendarController',
...
-                $scope.calendar = undefined;
-
                 if ($scope.isEmpty()) {
                     template.close('calendar');
                 } else {
+                    $scope.updateCalendars();
                     template.open('calendar', 'read-calendar');
-                }
+                    $scope.calendar = $scope.calendarCopy;
+                    angular.element(document.getElementsByClassName("sidebar")).scope().vm.calendar = $scope.calendar;
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1013]: - en 2 lignes:

let [variable]: IScope = angular.element(document.getElementById("sidebar")).scope();
puis utiliser variable

- vérifier que vm et calendar existent

File: src/main/resources/public/ts/controllers/controller.ts [hunk 2]
```diff
@@ -917,6 +917,7 @@ export const calendarController = ng.controller('CalendarController',
 
             $scope.editCalendar = function (calendar, event) {
                 $scope.calendar = calendar;
+                $scope.calendarCopy = angular.copy(calendar);
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:920]: changer le nom de variable pour que ca explique son utilité
$scope.calendarBeforeEdition par exemple

File: src/main/resources/public/ts/controllers/controller.ts [hunk 3]
```diff
@@ -1003,14 +1004,16 @@ export const calendarController = ng.controller('CalendarController',
             };
 
             $scope.cancelCalendarEdit = function () {
-                $scope.calendarCreationScreen = false;
-                $scope.calendar = undefined;
-
                 if ($scope.isEmpty()) {
                     template.close('calendar');
                 } else {
+                    $scope.updateCalendars();
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1010]: explique pk tu rajoutes cette méthode (en com de pr)

**Review by qpic-cgi — APPROVED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 4]
```diff
@@ -1003,14 +1004,19 @@ export const calendarController = ng.controller('CalendarController',
...
                     template.open('calendar', 'read-calendar');
-                }
+                    $scope.calendar = $scope.calendarBeforeEdition;
+                    let calendarElement : IScope = angular.element(document.getElementsByClassName("sidebar")).scope().vm.calendar;
+                    if(calendarElement){
+                        angular.element(document.getElementsByClassName("sidebar")).scope().vm.calendar = $scope.calendar;
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/controller.ts:1015]: pourquoi ne pas utiliser ta variable ??

File: src/main/resources/public/ts/controllers/controller.ts [hunk 5]
```diff
@@ -917,6 +917,7 @@ export const calendarController = ng.controller('CalendarController',
 
             $scope.editCalendar = function (calendar, event) {
                 $scope.calendar = calendar;
+                $scope.calendarBeforeEdition  = angular.copy(calendar);
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/controller.ts:920]: indentation

File: src/main/resources/public/ts/controllers/controller.ts [hunk 6]
```diff
@@ -1003,14 +1004,19 @@ export const calendarController = ng.controller('CalendarController',
...
-
                 if ($scope.isEmpty()) {
                     template.close('calendar');
                 } else {
+                    $scope.updateCalendars();
                     template.open('calendar', 'read-calendar');
-                }
+                    $scope.calendar = $scope.calendarBeforeEdition;
+                    let calendarElement : IScope = angular.element(document.getElementsByClassName("sidebar")).scope().vm.calendar;
+                    if(calendarElement){
```
Comment (JinDouang) [inline src/main/resources/public/ts/controllers/controller.ts:1014]: indentation espace du if

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 7]
```diff
@@ -1003,14 +1004,19 @@ export const calendarController = ng.controller('CalendarController',
...
-                $scope.calendarCreationScreen = false;
-                $scope.calendar = undefined;
-
                 if ($scope.isEmpty()) {
                     template.close('calendar');
                 } else {
+                    $scope.updateCalendars();
                     template.open('calendar', 'read-calendar');
+                    $scope.calendar = $scope.calendarBeforeEdition;
+                    let calendarElement = angular.element(document.getElementsByClassName("sidebar")).scope();
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1013]: donne un nom plus parlant, il faut qu'on comprenne que ca vient de la sidebar

File: src/main/resources/public/ts/controllers/controller.ts [hunk 8]
```diff
@@ -1003,14 +1004,19 @@ export const calendarController = ng.controller('CalendarController',
...
+                    if (calendarElement != undefined) {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1014]: il faut vérifier:
- si ton angular.element est défini
- si vm et calendar sont définis

Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1014]: pas besoin de != undefined

**Review by alicedraillard — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/controllers/controller.ts [hunk 9]
```diff
@@ -1003,14 +1004,19 @@ export const calendarController = ng.controller('CalendarController',
...
-                $scope.calendar = undefined;
-
                 if ($scope.isEmpty()) {
                     template.close('calendar');
                 } else {
+                    $scope.updateCalendars();
                     template.open('calendar', 'read-calendar');
+                    $scope.calendar = $scope.calendarBeforeEdition;
+                    let calendarElementSideBar = angular.element(document.getElementsByClassName("sidebar")).scope();
+                    if (calendarElementSideBar && calendarElementSideBar.vm.calendar) {
```
Comment (alicedraillard) [inline src/main/resources/public/ts/controllers/controller.ts:1014]: vérifie vm tout seul aussi sinon tu auras une exception

**Review by alicedraillard — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**